src/Schema/Column/AbstractColumnSchema.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Schema\Column;
+
+/**
+ * Represents the metadata of a column in a database table.
+ *
+ * It provides information about the column's name, type, size, precision, and other details.
+ *
+ * The `ColumnSchema` class is used to store and retrieve metadata about a column in a database table.
+ *
+ * It's typically used in conjunction with the TableSchema class, which represents the metadata of a database table as a
+ * whole.
+ *
+ * Here is an example of how to use the `ColumnSchema` class:
+ *
+ * ```php
+ * use Yiisoft\Db\Schema\ColumnSchema;
+ *
+ * $column = new ColumnSchema();
+ * $column->name('id');
+ * $column->allowNull(false);
+ * $column->dbType('int(11)');
+ * $column->phpType('integer');
+ * $column->type('integer');
+ * $column->defaultValue(0);
+ * $column->autoIncrement(true);
+ * $column->primaryKey(true);
+ * ``
+ */
+abstract class AbstractColumnSchema implements ColumnSchemaInterface
+{
+    private bool $allowNull = false;
+    private bool $autoIncrement = false;
+    private string|null $comment = null;
+    private bool $computed = false;
+    private string|null $dbType = null;
+    private mixed $defaultValue = null;
+    private array|null $enumValues = null;
+    private string|null $extra = null;
+    private bool $isPrimaryKey = false;
+    private string|null $phpType = null;
+    private int|null $precision = null;
+    private int|null $scale = null;
+    private int|null $size = null;
+    private string $type = '';
+    private bool $unsigned = false;
+
+    public function __construct(private string $name)
```

Comment:
How about add $type and $phptype params to constructor?

And use in child class:
parent::__costruct($name, SchemaInterface::TYPE_BIGINT, SchemaInterface::PHP_TYPE_BIGINT) etc

Reply:
Setters will be removed

Reply:
Done